packages/flutter/lib/src/material/navigation_bar.dart

```diff
@@ -459,24 +475,28 @@ class _NavigationDestinationBuilder extends StatelessWidget {
 
 class _IndicatorInkWell extends InkResponse {
   const _IndicatorInkWell({
-    super.child,
-    super.onTap,
+    super.key,
+    required this.labelPadding,
     super.customBorder,
+    super.onTap,
+    super.child,
   }) : super(
     containedInkWell: true,
     highlightColor: Colors.transparent,
   );
 
+  final double labelPadding;
+
   @override
   RectCallback? getRectCallback(RenderBox referenceBox) {
     final double indicatorOffsetX = referenceBox.size.width / 2;
-    const double indicatorOffsetY = 30.0;
+    final double indicatorOffsetY = referenceBox.size.height / 2 - labelPadding;
 
     return () {
       return Rect.fromCenter(
         center: Offset(indicatorOffsetX, indicatorOffsetY),
-        width: _kIndicatorHeight,
-        height: _kIndicatorWidth,
+        width: _kIndicatorWidth,
```

Comment:
Wow, sorry to have missed this in my review of https://github.com/flutter/flutter/pull/115499